### PR TITLE
Certification tests for Azure Blob Storage Binding

### LINKED ISSUE #956: Certification tests for Azure Blob Storage binding
## Release Note

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Adds certification tests for the Azure Blob Storage Binding and adds minor fixes to the binding itself

## Issue reference

closes #956 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/certification/bindings/azure/blobstorage/blobstorage_test.go [hunk 1]
```diff
@@ -0,0 +1,731 @@
+/*
+Copyright 2021 The Dapr Authors
```
Comment (artursouza) [inline tests/certification/bindings/azure/blobstorage/blobstorage_test.go:2]: nit: 2022 on all headers.

File: tests/certification/bindings/azure/blobstorage/blobstorage_test.go [hunk 2]
```diff
@@ -0,0 +1,731 @@
...
+	testCreateBlobFromFile := func(isBase64 bool) func(ctx flow.Context) error {
+		// uploads a blob from a file and optionally verifies the automatic base64 decoding option.
+		return func(ctx flow.Context) error {
+			client, clientErr := daprsdk.NewClientWithPort(fmt.Sprint(currentGRPCPort))
+			if clientErr != nil {
+				panic(clientErr)
+			}
+			defer client.Close()
+
+			dataBytes, dataErr := os.ReadFile("dapr.svg")
```
Comment (artursouza) [inline tests/certification/bindings/azure/blobstorage/blobstorage_test.go:251]: You don't need an external file for this. You can simply save a small byte array. Also, SVG seems to be a text file.

File: tests/certification/bindings/azure/blobstorage/go.mod [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	k8s.io/apimachinery v0.20.0 // indirect
+	k8s.io/client-go v0.20.0 // indirect
+	k8s.io/klog/v2 v2.4.0 // indirect
+	k8s.io/utils v0.0.0-20201110183641-67b214c5f920 // indirect
+	sigs.k8s.io/controller-runtime v0.7.0 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.0.2 // indirect
+	sigs.k8s.io/yaml v1.2.0 // indirect
+)
+
+replace github.com/dapr/components-contrib/tests/certification => ../../../
```
Comment (artursouza) [inline tests/certification/bindings/azure/blobstorage/go.mod:122]: You probably need to remove the replaces, not?

**Review by artursouza — CHANGES_REQUESTED** [review]

**Comment by artursouza** [comment]
It is pretty close to be merged. Made some small comments.

File: tests/certification/bindings/azure/blobstorage/blobstorage_test.go (hunk 1 above)
Comment (daixiang0) [reply tests/certification/bindings/azure/blobstorage/blobstorage_test.go:2]: I am confused about the year here, need to update all the time?

File: tests/certification/bindings/azure/blobstorage/go.mod (hunk 3 above)
Comment (berndverst) [reply tests/certification/bindings/azure/blobstorage/go.mod:122]: No these replace statements have to stay. This way we always run against the latest components contrib and certification framework in a given branch. This is consistent with the other certification tests.

**Comment by berndverst** [comment]
> It is pretty close to be merged. Made some small comments.

Done @artursouza

**Review by artursouza — DISMISSED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**